android/Staccato_AN/app/src/main/java/com/woowacourse/staccato/presentation/common/PhotoAttachFragment.kt

```diff
@@ -1,16 +1,118 @@
 package com.woowacourse.staccato.presentation.common
 
+import android.Manifest.permission.READ_EXTERNAL_STORAGE
+import android.Manifest.permission.READ_MEDIA_IMAGES
+import android.app.Activity.RESULT_OK
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
 import android.os.Bundle
+import android.provider.MediaStore
+import android.provider.Settings
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.google.android.material.snackbar.Snackbar
+import com.woowacourse.staccato.R
 import com.woowacourse.staccato.databinding.FragmentPhotoAttachBinding
+import com.woowacourse.staccato.presentation.util.showToast
+import com.woowacourse.staccato.presentation.visitcreation.OnUrisSelectedListener
 
-class PhotoAttachFragment : BottomSheetDialogFragment() {
+class PhotoAttachFragment : BottomSheetDialogFragment(), PhotoAttachHandler {
     private var _binding: FragmentPhotoAttachBinding? = null
     private val binding get() = _binding!!
 
+    private lateinit var uriSelectedListener: OnUrisSelectedListener
+    private lateinit var requestPermissionLauncher: ActivityResultLauncher<String>
+    private lateinit var galleryLauncher: ActivityResultLauncher<Intent>
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        initUriSelectedListner(context)
+        initRequestPermissionLauncher()
+        initGalleryLauncher()
+    }
+
+    private fun initUriSelectedListner(context: Context) {
+        if (context is OnUrisSelectedListener) {
+            uriSelectedListener = context
+        } else {
+            throw RuntimeException()
+        }
+    }
+
+    private fun initRequestPermissionLauncher() {
+        requestPermissionLauncher =
+            registerForActivityResult(ActivityResultContracts.RequestPermission()) { isPermissionGranted ->
+                if (isPermissionGranted) {
+                    launchGallery()
+                } else {
+                    showPermissionSnackBar()
+                }
+            }
+    }
+
+    private fun launchGallery() {
+        val intent =
+            Intent(Intent.ACTION_PICK).apply {
```

Comment:
setDataAndType와 putExtra는 Intent 자신을 반환하기 때문에 apply를 사용하지 않아도 된다고 알고 있습니다! 
 
<img width="612" alt="image" src="https://github.com/user-attachments/assets/d3be0e5a-0fca-47ae-87a9-f9f37f6755d7">

<img width="453" alt="image" src="https://github.com/user-attachments/assets/1c20fd93-5dd8-42e4-92f6-7021da381569">

<br>
<br>

 아래와 같이 수정해보는 것도 좋을 것 같아요~
```suggestion
            Intent(Intent.ACTION_PICK)
                .setDataAndType(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, "image/*")
                .putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true)
```